freshmaker/handlers/botas/botas_shipped_advisory.py

```diff
@@ -270,24 +306,107 @@ def handle(self, event):
+        return bundles_to_rebuild
 
-        # Reset context to db_event.
-        self.set_context(db_event)
+    def _handle_manual_rebuild(self, db_event):
+        """
+        Handle manual rebuild submitted by Release Driver for an advisory created by Botas
 
-        self.start_to_build_images(builds)
-        msg = f"Advisory {db_event.search_key}: Rebuilding " \
-              f"{len(db_event.builds.all())} bundle images."
-        db_event.transition(EventState.BUILDING, msg)
+        :param db_event: database event that represent rebuild event
+        :rtype: list
+        :return: list of advisories that should be rebuilt
+        """
+        default_bundle_data = {
+            'nvr': '',
+            'update': {},
+            'append': {},
+            'pullspecs': [],
+        }
+        # map with old pullspec and bundle info
+        old_pullspec_to_bundle_map = dict()
+        # get all replaced pullspecs from provided 'bundle_images'
+        for bundle_nvr in self.event.bundle_images:
+            artifact_build = db.session.query(ArtifactBuild).filter(
+                ArtifactBuild.rebuilt_nvr == bundle_nvr,
+                ArtifactBuild.type == ArtifactType.IMAGE.value,
+            ).one_or_none()
+            pullspec_overrides = artifact_build.bundle_pullspec_overrides
+            for pullspec in pullspec_overrides:
+                old_pullspec = pullspec.get('_old', None)
+                if old_pullspec is not None:
+                    old_pullspec_to_bundle_map.setdefault(
+                        old_pullspec, copy.deepcopy(default_bundle_data))
+                    # bundle will now contain new pullspec overrides
+                    old_pullspec_to_bundle_map[old_pullspec]['pullspecs'] = pullspec_overrides
+                    break
+
+        if not old_pullspec_to_bundle_map:
+            msg = 'None of the bundle images have old pullspecs overrides in FM database'
+            log.warning(msg)
+            db_event.transition(EventState.SKIPPED, msg)
+            return []
 
-        return []
+        # Unauthenticated koji session to fetch build info of bundles
+        koji_api = KojiService(conf.koji_profile)
+        rebuild_nvr_to_bundle_map = dict()
+        # compare replaced pullspecs with pullspecs in 'container_images' and
+        # create map for bundles that should be rebuilt with their nvrs
+        for container_image_nvr in self.container_images:
+            # Fetch buildinfo
+            buildinfo = koji_api.get_build(container_image_nvr)
+            # Get the original pullspecs
+            pullspecs = (
+                buildinfo.get('extra', {})
+                         .get('image', {})
+                         .get('operator_manifests', {})
+                         .get('related_images', {})
+                         .get('pullspecs', [])
+            )
+            for pullspec in pullspecs:
+                if pullspec.get('new') in old_pullspec_to_bundle_map:
```

Comment:
Optional: This will make it easier to read.
```suggestion
                if pullspec.get('new') not in old_pullspec_to_bundle_map:
                    continue
```